src/fcntl.rs

```diff
@@ -305,6 +308,43 @@ pub fn flock(fd: RawFd, arg: FlockArg) -> Result<()> {
     Errno::result(res).map(drop)
 }
 
+
+#[cfg(any(target_os = "linux", target_os = "android"))]
+bitflags! {
+    pub struct CopyFileRangeFlags: c_int {
+        const NONE = 0 as c_int;
+    }
+}
+
+/// Copy a range of data from one file to another
+///
+/// The `copy_file_range` system call performs an in-kernel copy between
+/// file descriptor `fd_in` and `fd_out` without the additional cost of transferring data
+/// from the kernel to user space and then back into the kernel.  It
+/// copies up to `len` bytes of data from file descriptor `fd_in` to file
+/// descriptor `fd_out`, overwriting any data that exists within the
+/// requested range of the target file.
+///
+/// The flags argument is currently unused, and is provided to allow for future
+/// extensions and currently must be set to `CopyFileRangeFlags::empty()`.
+///
+/// On completion the number of bytes actually copied will be returned.
+#[cfg(any(target_os = "android", target_os = "linux"))]
+pub fn copy_file_range(
+    fd_in: RawFd, off_in: Option<&mut libc::loff_t>,
+    fd_out: RawFd, off_out: Option<&mut libc::loff_t>,
+    len: usize,
+    _flags: CopyFileRangeFlags) -> Result<usize> {
```

Comment:
I must admit I'm not enthusiastic about the `flags` argument.  It currently provides no benefit to the user, and there's no guarantee that Linux will ever use it.  But it does require a lot of typing.  Maybe we should omit it.  Or we could at least reduce the typing by changing the type to `Into<CopyFileRangeFlags>` so the user can simply supply `0`.  What do you think?

Reply:
Removed the flags.